packages/yarnpkg-fslib/sources/ZipFS.ts

```diff
@@ -815,21 +815,17 @@ export class ZipFS extends BasePortableFakeFS {
 
       resolvedP = ppath.resolve(parentP, ppath.basename(resolvedP));
       if (!resolveLastComponent || this.symlinkCount === 0)
-        break;
+        return resolvedP;
 
       const index = this.libzip.name.locate(this.zip, resolvedP.slice(1));
       if (index === -1)
-        break;
+        return resolvedP;
+      if (!this.isSymbolicLink(index))
+        return resolvedP;
```

Comment:
You could combine these into a single if statement. I would make it a suggestion but GitHub wont let me.

Reply:
Done in [`d4657dc` (#4103)](https://github.com/yarnpkg/berry/pull/4103/commits/d4657dc4a94dfb1552ee4eb1b69c40f1997687e3)